pkg/docker/config/config.go

```diff
@@ -606,6 +636,45 @@ func findAuthentication(registry, path string, legacyFormat bool) (types.DockerA
 	return types.DockerAuthConfig{}, nil
 }
 
+func findAuthenticationForRef(auths dockerConfigFile, ref reference.Named) (types.DockerAuthConfig, error) {
+	for _, registry := range pathsForRef(ref) {
+		normalizedAuth := normalizeRegistry(registry)
```

Comment:
Do we need this at all? It’s tempting to steer users towards `docker.io/foo/bar`, `docker.io/foo`, `docker.io`.

(*NOTE*: verify K8s can accept those values. I haven’t now checked. Has anyone looked into this already?)

Reply:
I thought about still stripping the `https://` prefix from registries as some sort of feature. What would be the reason to drop that?